hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/HttpAttributeEnricherTest.java

```diff
@@ -195,30 +195,6 @@ public void testDecodeQueryParams() {
     assertEquals(List.of("hello=world"), locationParam.getValueList());
   }
 
-  @Test
-  public void testDecodeQueryParamsInvalidInput() {
-    //Try putting invalid encoded string in both query param key and value.
-    Event e = createMockEvent();
-    when(e.getHttp())
-        .thenReturn(org.hypertrace.core.datamodel.eventfields.http.Http.newBuilder().setRequest(
-            Request.newBuilder()
-                .setUrl("http://hypertrace.org/users?action=check%!mout&loca%.Ption=hello%3Dworld")
```

Comment:
Constructing a `URI` object for such invalid URIs throws a `URISyntaxException` whereas constructing a `URL` object from this URL was working fine. This is weird as this doesn't follow the fact that every URL is a URI. Anyways, with URI we can't process such invalid inputs. 

Is there a strong requirement for processing such invalid inputs? I'm asking because with URI we can't process such invalid inputs.

Reply:
I remember @anujgoyal1 added this unit test for some encoding related issue. @anujgoyal1 Can you please take a look at it?

Reply:
@GurtejSohi In we see such inputs for real in the prod, not sure what's the ideal thing to do here. @skjindal93  what do you suggest?